libqtile/widget/battery_monitor.py

```diff
@@ -0,0 +1,394 @@
+# Copyright (c) 2011 matt
+# Copyright (c) 2011 Paul Colomiets
+# Copyright (c) 2011-2014 Tycho Andersen
+# Copyright (c) 2012 dmpayton
+# Copyright (c) 2012 hbc
+# Copyright (c) 2012 Tim Neumann
+# Copyright (c) 2012 uberj
+# Copyright (c) 2012-2013 Craig Barnes
+# Copyright (c) 2013 Tao Sauvage
+# Copyright (c) 2014 Sean Vig
+# Copyright (c) 2014 dequis
+# Copyright (c) 2014 Sebastien Blot
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import os
+import re
+import platform
+from enum import Enum
+from subprocess import check_output, CalledProcessError
+from libqtile import bar
+from . import base
+from .. import images
+
+
+class State(Enum):
+    CHARGING = 0
+    DISCHARGING = 1
+    FULL = 2
+    EMPTY = 3
+    UNKNOWN = 4
+
+
+class _Battery(object):
```

Comment:
I actually like this a lot more as an interface that we could port the other battery over to. It would be awesome if you could also add some type annotations to it!

It'd be good to make this an `abc.ABCMeta` abstract class and all of the not implemented methods as abstract methods.

Reply:
Done

Reply:
I wrote the data type in the comment. If you want them in some specific format, please give me an example.